imagetest/testworkflow.go

```diff
@@ -99,7 +99,75 @@ func (t *TestWorkflow) CreateTestVM(name string) (*TestVM, error) {
 	return &TestVM{name: vmname, testWorkflow: t}, nil
 }
 
-func (t *TestWorkflow) appendCreateVMStep(name, hostname string) (*daisy.Step, error) {
+// CreateTestVMWithCustomNetwork creates the necessary steps to create a VM with  the specified name and custom network to the workflow.
+func (t *TestWorkflow) CreateTestVMWithCustomNetwork(name, networkName,
+	subnetworkName, rangeName, primaryIPRange, secondaryIPRange, aliasIPRange string) (*TestVM, error) {
+	parts := strings.Split(name, ".")
+	vmname := strings.ReplaceAll(parts[0], "_", "-")
+
+	createDisksStep, err := t.appendCreateDisksStep(vmname)
+	if err != nil {
+		return nil, err
+	}
+
+	addCreateNetworkStep, err := t.addCreateNetworkStep(vmname, networkName)
+	if err != nil {
+		return nil, err
+	}
+
+	addCreateSubnetworkStep, err := t.addCreateSubnetworkStep(vmname, networkName, subnetworkName, rangeName, primaryIPRange, secondaryIPRange)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := t.wf.AddDependency(addCreateSubnetworkStep, addCreateNetworkStep); err != nil {
+		return nil, err
+	}
+
+	networkinterface := []*compute.NetworkInterface{
+		{
+			Network:    networkName,
+			Subnetwork: subnetworkName,
+			AccessConfigs: []*compute.AccessConfig{
+				{
+					Type: "ONE_TO_ONE_NAT",
+				},
+			},
+			AliasIpRanges: []*compute.AliasIpRange{
+				{
+					IpCidrRange:         aliasIPRange,
+					SubnetworkRangeName: rangeName,
+				},
+			},
+		},
+	}
+	// createDisksStep doesn't depend on any other steps.
+	createVMStep, err := t.appendCreateVMStep(vmname, name, networkinterface)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := t.wf.AddDependency(createVMStep, addCreateSubnetworkStep); err != nil {
+		return nil, err
+	}
+
+	if err := t.wf.AddDependency(createVMStep, createDisksStep); err != nil {
+		return nil, err
+	}
+
+	waitStep, err := t.addWaitStep(vmname, vmname, false)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := t.wf.AddDependency(waitStep, createVMStep); err != nil {
+		return nil, err
+	}
+
+	return &TestVM{name: vmname, testWorkflow: t}, nil
+}
+
+func (t *TestWorkflow) appendCreateVMStep(name, hostname string, networkInterface []*compute.NetworkInterface) (*daisy.Step, error) {
```

Comment:
yeah now you're making it so the network is a required parameter, forcing you to pass nil to get the default network. the paradigm in these fixtures is to fully populate the steps with fixtures over time, then finalize them. you can add these network details to the step *after* it has been created in the workflow.